src/sentry/nodestore/riak/client.py

```diff
@@ -248,8 +259,10 @@ def urlopen(self, method, path, **kwargs):
                     self.force_revive()
 
                 conn = self.strategy.next(self.connections)  # NOQA
+                if headers is not None:
+                    headers = dict(conn.headers, **headers)
```

Comment:
We need to merge the user supplied headers that are being set per-request, with the connection level default headers.

The default behavior of urllib3 is to only use the connection level defaults if there are no others.